TYPO3.Flow/Classes/TYPO3/Flow/Mvc/ViewConfigurationManager.php

```diff
@@ -83,13 +83,15 @@ public function getViewConfiguration(ActionRequest $request)
             }
 
             usort($matchingConfigurations, function ($configuration1, $configuration2) {
-                return $configuration1['weight'] > $configuration2['weight'];
+                return $configuration1['weight'] < $configuration2['weight'];
             });
-
-            $viewConfiguration = [];
-            foreach ($matchingConfigurations as $key => $matchingConfiguration) {
-                $viewConfiguration = Arrays::arrayMergeRecursiveOverrule($viewConfiguration, $matchingConfiguration['configuration']);
+            
+            if (count($matchingConfigurations) > 0) {
+                $viewConfiguration = array_shift($matchingConfigurations)['configuration'];
+            } else {
+                $viewConfiguration = [];
             }
+
```

Comment:
Nitpick: Without the merging the whole block could be simplified IMO along those lines:
```php
$highestWeight = -1;
foreach ($configurations as $order => $configuration) {
    $requestMatcher->resetWeight();
    if (!isset($configuration['requestFilter']) && $order > $highestWeight) {
        $weight = $order;
    } else {
        $result = $this->eelEvaluator->evaluate($configuration['requestFilter'], $context);
        if ($result === false) {
            continue;
        }
        $weight = $requestMatcher->getWeight() + $order;
    }
    if ($weight > $highestWeight) {
        $viewConfiguration = $configuration;
        $highestWeight = $weight;
    }
}
```